deploy/docker/cp-edge/sync-routes.py

```diff
@@ -412,17 +463,35 @@ def get_service_list(pod_id, pod_run_id, pod_ip):
         print('EDGE service port: ' + str(edge_service_port))
         print('EDGE service ip: ' + edge_service_external_ip)
 
-# From each pod with "job-type=Service"  we shall take:
+# From each pod with a container, which has endpoints ("job-type=Service" or container's environment
+# has a parameter from SYSTEM_ENDPOINTS) we shall take:
 # -- PodIP
 # -- PodID
 # -- N entries by a template
 # --- svc-port-N
 # --- svc-path-N
-pods = Pod.objects(kube_api).filter(selector={'job-type': 'Service'})\
-                            .filter(field_selector={"status.phase": "Running"})
+
+def load_pods_for_runs_with_endpoints():
+        pods_with_endpoints = []
+        all_pipeline_pods = Pod.objects(kube_api).filter(selector={'type': 'pipeline'})\
+                                                 .filter(field_selector={"status.phase": "Running"})
+        for pod in all_pipeline_pods.response['items']:
+                labels = pod['metadata']['labels']
+                if 'job-type' in labels and labels['job-type'] == 'Service':
+                        pods_with_endpoints.append(pod)
+                        continue
+                pipeline_env_parameters = pod['spec']['containers'][0]['env']
```

Comment:
Should we add check 'spec' and 'containers' for existence and `pod['spec']['containers'][0]` for not null?

Reply:
Done, perform additional checks, that response structure is as expected